addons/mail/static/src/components/attachment_viewer/attachment_viewer.js

```diff
@@ -137,7 +137,10 @@ export class AttachmentViewer extends Component {
      */
     _download() {
         const id = this.attachmentViewer.attachment.id;
-        this.env.services.navigate(`/web/content/ir.attachment/${id}/datas`, { download: true });
+        let downloadLink = document.createElement('a');
+        downloadLink.setAttribute('href', `/web/content/ir.attachment/${id}/datas`);
+        downloadLink.setAttribute('download','');
+        downloadLink.click();
```

Comment:
Why do we have a duplicate download code? This one should just call the one from attachment model no?

Reply:
The original code is not calling attachment model, it is duplicated and I just remain the same for that..

Reply:
It's usually good to avoid useless diff, but in this case it is actually more diff as you have to introduce all this new code twice.

So I would take the opportunity to clean it and have only one method to download.